### PR TITLE
Flag gas related MUSTFIX

### PR DESCRIPTION
**Pull request by lxfind**
I looked at some of the TODOs and I believe some of them are MUSTFIX. Let me know if I got them wrong.

### CODE REVIEW

File: crates/sui-types/src/gas.rs [hunk 1]
```diff
@@ -239,9 +239,6 @@ pub struct SuiCostTable {
     /// Unit cost of a byte in the storage. This will be used both for charging for
     /// new storage as well as rebating for deleting storage. That is, we expect users to
     /// get full refund on the object storage when it's deleted.
-    /// TODO: We should introduce a flat fee on storage that does not get refunded even
-    /// when objects are deleted. This cost covers the cost of storing transaction metadata
```
Comment (oxade) [inline crates/sui-types/src/gas.rs:243]: Yes this comment should not be there anymore; it’s leftover from when we tried flat fees.

Comment (oxade) [reply crates/sui-types/src/gas.rs:243]: Sorry I commented on the wrong change.

File: crates/sui-types/src/gas.rs [hunk 2]
```diff
@@ -381,7 +378,7 @@ impl<'a> SuiGasStatus<'a> {
     }
 
     pub fn charge_vm_gas(&mut self) -> Result<(), ExecutionError> {
-        // Disable flat fee for now
+        // MUSTFIX: Disable flat fee for now. Do we ever enable this?
```
Comment (oxade) [inline crates/sui-types/src/gas.rs:381]: This is what we should remove @lxfind 
Re: this [comment](https://github.com/MystenLabs/sui/pull/9282/files#r1136101848)

File: crates/sui-types/src/gas.rs [hunk 3]
```diff
@@ -454,7 +445,7 @@ impl<'a> SuiGasStatus<'a> {
     pub fn summary(&self) -> GasCostSummary {
         let remaining_gas = self.gas_status.remaining_gas();
         let storage_cost = self.storage_gas_units;
-        // TODO: handle underflow how?
+        // MUSTFIX: handle underflow how?
```
Comment (dariorussi) [inline crates/sui-types/src/gas.rs:448]: not clear this should happen but good to have a mustffix to review rather than a todo

**Review by dariorussi — APPROVED** [review]
as I asked should we have issues opened you think?
It makes no difference to me and those are issues on my plate and working on them.
As for the flat fee we are charging that through buckets, but that overall needs to be reviewed together with gas smashing and making sure we are safe

File: crates/sui-core/src/transaction_input_checker.rs [hunk 4]
```diff
@@ -190,7 +190,7 @@ async fn check_gas(
         let cost_table = SuiCostTable::new(protocol_config);
         let storage_gas_price = protocol_config.storage_gas_price();
 
-        // TODO: We should revisit how we compute gas price and compare to gas budget.
+        // MUSTFIX: We should revisit how we compute gas price and compare to gas budget.
```
Comment (emmazzz) [inline crates/sui-core/src/transaction_input_checker.rs:193]: We compute the max of computation_gas_price and storage_gas_price here too:https://github.com/MystenLabs/sui/blob/28526f64b170d2dbc4bf1c9660fad05858f8b6b8/crates/sui-types/src/gas.rs#L368
which seems to be used for programmable txns.

Comment (dariorussi) [reply crates/sui-core/src/transaction_input_checker.rs:193]: the plan is to remove those and to use computation for computation and gas for gas...
just fyi

Comment (emmazzz) [reply crates/sui-core/src/transaction_input_checker.rs:193]: yup just pointing out another place where we should considering adding a MUSTFIX...

**Pull request merged**